app/scripts/app-init.js

```diff
@@ -132,12 +132,15 @@ chrome.runtime.onMessage.addListener(() => {
  * MAIN world injection does not work properly via manifest
  * https://bugs.chromium.org/p/chromium/issues/detail?id=634381
  */
-chrome.scripting.registerContentScripts([
-  {
-    id: 'inpage',
-    matches: ['file://*/*', 'http://*/*', 'https://*/*'],
-    js: ['inpage.js'],
-    runAt: 'document_start',
-    world: 'MAIN',
-  },
-]);
+// eslint-disable-next-line no-undef
+self.addEventListener('install', () => {
```

Comment:
This event only fires when a new service worker is installed. It will not necessarily fire when we terminate the service worker since we have our keep-alive logic. It also does not seem to fire upon refreshing the page. 

chrome.runtime.onConnect fires when a connection is made with the contentscript. maybe we can check if the `inpage.js` script has been loaded here.